sympy/functions/combinatorial/numbers.py

```diff
@@ -2292,3 +2292,149 @@ def ok(x):
+    [1, 1, 1, 2, 2, 3, 4, 5]
+    >>> padovan.find_padovan_numbers_in_range(2,39)
+    [2, 2, 3, 4, 5, 7, 9, 12, 16, 21, 28, 37]
+
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Padovan_sequence
+    .. [2] https://mathworld.wolfram.com/PadovanSequence.html
+
+    """
+
+    @staticmethod
+    def is_padovan(n):
+        try:
+            n = as_int(n)
+        except ValueError:
+            return False
+        if n>0:
+            if n == 1 :
+                return True
+
+            sn0 = 1
+            sn1 = 1
+            sn2 = 1
+            while sn2 < n :
+                b = sn0 + sn1
+                sn0 = sn1
+                sn1 = sn2
+                sn2 = b
+
+            if Eq(sn2, n, evaluate=True):
+                return True
+            else:
+                return False
+
+        if n<=0:
+            if n in(0,-1):
+                return True
+
+            sn0 = 1
+            sn1 = 0
+            sn2 = 1
+            while sn2 > n :
+                b = sn0 - sn2
+                sn0 = sn1
+                sn1 = sn2
+                sn2 = b
+
+            if Eq(sn2, n, evaluate=True):
+                return True
+            else:
+                return False
+
+    @staticmethod
+    def find_first_n_padovan_numbers(n):
+        # Returns first n non-negative numbers from the padovan sequence
+        try:
+            n = as_int(n)
+        except ValueError:
+            raise ValueError('The provided number must be a positive integer')
+        if n < 0:
+            raise ValueError('The provided number must be a positive integer')
+        padovan = [1]
+        if n > 1:
+            padovan.append(1)
+
+        sn0 = 1
+        sn1 = 1
+        sn2 = 1
+        i = 2
+        while i < n :
+            padovan.append(sn2)
+            b = sn0 + sn1
+            sn0 = sn1
+            sn1 = sn2
+            sn2 = b
+            i = i + 1
+
+        return padovan
+
+    @staticmethod
+    def find_padovan_numbers_in_range(x, y):
+        if 0 <= x <= y:
+            padovan = list()
+            if x <=1 <= y:
+                padovan.extend([1,1])
+            sn0 = 1
+            sn1 = 1
+            sn2 = 1
+            i = 2
+            while sn2 <= y:
+                if sn2 >= x:
+                    padovan.append(sn2)
+                b = sn0 + sn1
+                sn0 = sn1
+                sn1 = sn2
+                sn2 = b
+                i = i + 1
+
+            return padovan
+
+        else:
+            raise ValueError('The provided range is not valid. This condition should satisfy x <= y')
+
+    @staticmethod
+    @recurrence_memo([S.One, S.One, S.One])
+    def _padovan(n, prev):
+        if n>0:
+            return (prev[-3] + prev[-2])
+
+    @classmethod
+    def eval(cls, n):
+        try:
+            n = as_int(n)
+        except ValueError:
+            raise ValueError('The provided number must be a integer')
```

Comment:
What if it's a symbol?

Reply:
> What if it's a symbol?

Yeah it is a subclass of functions ,it should not only accept integers but symbols too!
How would you suggest to deal with this ,I'll make appropriate changes in the next commit !!

Reply:
It's enough to return `None` from here I think rather raising `ValueError`.

Reply:
Thanks @oscarbenjamin .I have made the change you suggested .